docs/_posts/2024-07-18-Run-your-pandas-workloads-on-cluster.md

```diff
@@ -0,0 +1,213 @@
+    result['store_and_fwd_flag'] = df['store_and_fwd_flag'].replace({'N': 'No', 'Y': 'Yes'})
+    result["total_amount_rounded"] = df["total_amount"].round(1)
+    execute(result)
+    </code></pre>
+
+  ```
+</details>
+
+### Results
+
+We can observe that by increasing the number of nodes in the cluster, Modin could handle the workload
+effectively with 4 nodes or more. Significant performance improvements were evident as we scaled
+from 4 to 32 nodes. Given that Modin operates efficiently with 4 nodes and both pandas(as pandas does not offer horizontal scaling) and Modin
+struggle with fewer than 3 nodes, it suggests the workload demands nearly 4 times the memory (32GB) of
+a single t3.2xlarge instance.
+
+To provide a fair comparison of the performance gains offered by Modin over pandas,
+we decided to benchmark the script on an instance with over 120 GB of memory.
+We chose an r6a.4xlarge instance, which boasts 128GiB of RAM.
+
+With pandas, the script took almost 400 seconds to execute, which serves as our baseline.
+
+### Performance on a Single Node
+
+The graph reveals an interesting twist: Modin on a single node starts off slower than pandas
+when using 4 workers or fewer. This could be because the overhead of distributing data and spinning up
+the Ray cluster doesn't pay off with so few workers. It's like trying to split chores among two people
+when you really need a full team. But here's where it gets exciting: crank up the CPU count
+(even just to 8), and Modin shifts into high gear, outperforming pandas even on a single node.
+Talk about a comeback!
+
+<img src="imgs/blog_post_4/Modin_single_node.svg" alt="Perf Results single node"  style="display: block; margin-left: auto; margin-right: auto;">
+
+
+### Performance on Scaling Nodes to 32
+
+The performance continues to soar as we add more nodes to the cluster. With a 32-node cluster,
+utilizing all 16 CPUs in each node, the script executes in a fraction of the time it takes on a single node.
+This results in a dramatic performance boost, showcasing the impressive scalability of Modin.
+
+<img  src="imgs/blog_post_4/Modin_multiple_nodes.svg" alt="Perf Results multinode"  style="display: block; margin-left: auto; margin-right: auto;">
+
+<div style="border: 1px solid #ddd; padding: 10px; margin-top: 20px; background-color: #f9f9f9;">
+<strong>Disclaimer:</strong> The workload demonstrated in this post is designed to be
+embarrassingly parallel to illustrate Modin's horizontal scaling capabilities.
+Be aware that it is possible results may not show positive scaling in a cluster environment if
+operations require data transfer across nodes.
+</div>
+
+## Further Reading
+
+For more information on using Modin, please refer to the [official documentation](https://modin.readthedocs.io/en/stable/getting_started/using_modin/using_modin.html).
+
+#### Appendix
+
+All performance measurements for this post were made on an AWS r6a.4xlarge instance.
+
+- Modin version: 0.32.0
```

Comment:
The numbers currently in the plots have no t been measured with modin 0.32.0 the numbers would be updated on release

Reply:
Have updated the  plots with measurements from modin 0.32.0.